checkov/sast/runner.py

```diff
@@ -0,0 +1,147 @@
+from semgrep.constants import OutputFormat, RuleSeverity
+from typing import Collection, List, Set, Dict, Tuple, TYPE_CHECKING
+from io import StringIO
+from pathlib import Path
+
+
+if TYPE_CHECKING:
+    import semgrep.output_from_core as core
+    from semgrep.rule_match import RuleMatchMap, RuleMatch
+    from semgrep.target_manager import FileTargetingLog
+    from semgrep.profile_manager import ProfileManager
+    from semgrep.profiling import ProfilingData
+    from semgrep.parsing_data import ParsingData
+    from semgrep.error import SemgrepError
+    from semgrep.rule import Rule
+
+
+logger = logging.getLogger(__name__)
+
+
+SEMGREP_SEVERITY_TO_CHECKOV_SEVERITY = {
+    RuleSeverity.ERROR: 'HIGH',
+    RuleSeverity.WARNING: 'MEDIUM',
+    RuleSeverity.INFO: 'LOW',
+}
+
+
+@dataclass
+class SemgrepOutput:
+    matches: RuleMatchMap
+    errors: List[SemgrepError]
+    all_targets: Set[Path]
+    renamed_targets: Set[Path]
+    target_manager_ignore_log: FileTargetingLog
+    filtered_rules: List[Rule]
+    profiler: ProfileManager
+    profiling_data: ProfilingData
+    parsing_data: ParsingData
+    explanations: List[core.MatchingExplanation]
+    shown_severities: Collection[RuleSeverity]
+    target_manager_lockfile_scan_info: Dict[str, int]
+
+
+class Runner():
+    check_type = CheckType.SAST  # noqa: CCE003  # a static attribute
+
+    def run(self, root_folder: str | None, external_checks_dir: list[str] | None = None, files: list[str] | None = None,
+            runner_filter: RunnerFilter | None = None, collect_skip_comments: bool = True) -> Report:
+        StringIO()
+        output_settings = OutputSettings(output_format=OutputFormat.JSON)
+        output_handler = OutputHandler(output_settings)
+
+        registry.load_checks(runner_filter.sast_languages)
+        if external_checks_dir:
+            for external_checks in external_checks_dir:
+                registry.load_external_checks(external_checks, runner_filter.sast_languages)
+
+        if root_folder:
+            targets = [root_folder]
+        if files:
+            targets = files
+        config = registry.checks
+        if not config:
+            logger.warning('no valid checks')
+            return Report(self.check_type)
+
+        semgrep_output = Runner._get_semgrep_output(targets=targets, config=config, output_handler=output_handler)
+        report = self._get_report(semgrep_output.matches)
+        return report
+
+    @staticmethod
+    def _get_semgrep_output(targets: List[str], config: List[str], output_handler: OutputHandler) -> SemgrepOutput:
+        (filtered_matches_by_rule,
+         semgrep_errors,
+         all_targets,
+         renamed_targets,
+         target_manager_ignore_log,
+         filtered_rules,
+         profiler,
+         profiling_data,
+         parsing_data,
+         explanations,
+         shown_severities,
+         target_manager_lockfile_scan_info) = run_semgrep(output_handler=output_handler, target=targets,
```

Comment:
seems like `run_semgrep` returns much more data, shouldn't we explicitly ignore the ones we don't need, so we won't accidentally get weird values for the ones we do need (in terms of the param order)
![image](https://user-images.githubusercontent.com/52502521/216823059-3332b24c-3117-4d2b-889d-bf56fc38ca70.png)